Add Issuer as regular packet during signing.

thorhs opened the pull request:
RPM in RHEL 8 and 9 are unable to find the signing key id when signed with this the rpm crate.  Newer versions of RPM do not have this issue.

By adding a second Issuer packet which is not critical works around the old RPM deficiency.

Closes #265
<!---
Thank you for submitting a PR to the rust rpm implementation!

Commits should be distinct and have a clear purpose, with messages
which explain to the reviewer WHAT changed, and WHY it had to change
and how the WHAT and the WHY tie together.

At the bottom of your commit messages, add a line "Closes #IssueNumber)"
for any issues resolved by the commit, substituting in an actual issue number.
This will automatically close the issue once the PR is merged and creates a
cross reference.

If things are still WIP or feedback on particular impl details
are wanted, state them here or leave comments below.
-->

### 📜 Checklist

- [x] Commits are cleanly separated and have useful messages
- [x] A changelog entry or entries has been added to CHANGELOG.md
- [x] Documentation is thorough
- [x] Test coverage is excellent and passes
- [x] Works when tests are run `--all-features` enabled

thorhs commented:
Just to be sure it does not get overlooked.  I am able to verify the signature of the RPM packages signed using the rpm crate on older versions of RPM.  The fix was to add the regular issuer packet, but I have no idea if it will cause issues for others reading RPM packages.

dralley commented on src/rpm/signature/pgp.rs:
```diff
@@ -66,6 +66,11 @@ where
+            .hashed_subpackets
+            .push(Subpacket::regular(SubpacketData::Issuer(
+                self.secret_key.key_id(),
+            )));
```
Ideally we want to match what RPM is doing as closely as possible.  If standard RPM doesn't mark these subpackets critical and doesn't duplicate them (which based on the outputs you posted on the issue seems to be the case), then we probably shouldn't be doing that either.  

Could you try having just one "regular" copy of both the issuer and signature creation time?

Also, I think the issuer fingerprint can be handled with this:

```
     sig_cfg
            .hashed_subpackets
            .push(Subpacket::regular(SubpacketData::IssuerFingerprint(
                self.secret_key.fingerprint(),
            )));
```

thorhs replied:
I've added the fingerprint as well, initially I wanted to do the minimum change to get signing to work.

thorhs commented:
@dralley I've refactored the commits a bit.  The first commit is refactoring how the test harness installs packages.  The differences between the distros were becoming troublesome so I moved it to the front of the test run and added some distro detection logic.

The second one adds the additional distros to test.

The third changes from critical to regular packets.

And the last one adds issuer fingerprint.

I decided to split them up for easier review, feel free to squash them.

dralley reviewed (approved):
Great work, thanks!
